feat(sqlmigration): squash all migrations into one

therealpandey opened the pull request:
## 📄 Summary

squash all migrations into 100 series

<!-- CURSOR_SUMMARY -->
---

> [!NOTE]
> Consolidates SQL migration system and adds operational tooling.
> 
> - Squashes legacy `pkg/sqlmigration` files into a single v100 series via `s100sqlmigration`; updates providers to use `NewSQLMigrationProviderFactories(sqlstore, sqlschema)`
> - New `metastore` CLI with `migrate sync up` to run schema migrations; registered in community and enterprise mains
> - Refactors server wiring to pass `sqlstoreProviderFactories`/`sqlschemaProviderFactories`; enterprise version adds Postgres providers and hooks
> - Postgres SQL schema tweaks: new operator support flags, better not-found error wrapping, fixed index column ordering/naming
> - Removes EE Postgres dialect implementation and `Dialect()` exposure from SQL store provider
> - Cloud integrations upsert now conflicts on `id` only
> - Updates docs to describe migration pack layout (series-based)
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit f35db14ef84523bccbb624dca4ef7464cec82d92. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>
<!-- /CURSOR_SUMMARY -->

grandwizard28 commented:
Closes https://github.com/SigNoz/platform-pod/issues/706

therealpandey committed:
docs(contributing): modify sql docs

therealpandey commented on pkg/sqlmigration/s100sqlmigration/100_v100.go:
```diff
@@ -0,0 +1,930 @@
+			TableName:   sqlschema.TableName("auth_token"),
+			ColumnNames: []sqlschema.ColumnName{"refresh_token"},
+		},
+	}
```
Good job!

swapnil-signoz replied:
@therealpandey I think this will cause problem. The older index (id, provider, org_id) did not add any value from uniqueness point of view as presence of `id` will always make the combination unique any way.

The new unique index without `id` column in it will be problematic as we do support multiple accounts from same provider to be connected with single Org e.g. we can connect multiple AWS accounts to be monitored for a single Org.
I think the intended (at the time of development) unique index was `(account_id, provider, org_id)`

I think this was missed and never caught the attention because `id` always made the combination unique and never caused conflict error. What we ideally want to restrict is not able to have same account from same cloud provider attached with a single ord_id; which is a bug right now WITH or WITHOUT this change.

cc: @piyushsingariya

therealpandey commented on pkg/sqlmigration/s100sqlmigration/100_v100.go:
```diff
@@ -0,0 +1,930 @@
+			TableName:   sqlschema.TableName("auth_token"),
+			ColumnNames: []sqlschema.ColumnName{"refresh_token"},
+		},
+	}
```
Good job!

swapnil-signoz commented on pkg/query-service/app/cloudintegrations/accountsRepo.go:
```diff
@@ -176,7 +176,7 @@ func (r *cloudProviderAccountsSQLRepository) upsert(
 	if len(onConflictSetStmts) > 0 {
 		onConflictClause = fmt.Sprintf(
-			"conflict(id, provider, org_id) do update SET\n%s",
+			"conflict(id) do update SET\n%s",
```
The change makes sense as other two columns are redundant. But please check my comment [here](https://github.com/SigNoz/signoz/pull/8564#discussion_r2657912208).

therealpandey commented on cmd/community/metastore.go:
```diff
@@ -0,0 +1,16 @@
+package main
```
@therealpandey please add seed data

therealpandey committed:
feat(sqlmigration): add s100sqlmigration package with v100 squash

Lift the squashed s100sqlmigration/100_v100.go and its test from #8564
onto current main. The file is a self-contained DDL definition of the
target schema captured via pkg/sqlschema; it compiles and tests pass
against main's pkg/sqlschema as-is.

Not yet wired into signoz.NewSQLMigrationProviderFactories — landing
the package on its own so the broader squash work can be staged
incrementally without dragging in the rest of #8564's divergent state.

refs SigNoz/platform-pod#706

therealpandey committed:
feat(sqlmigration): encode end-state schema of 000-082 in v100

Extend v100 to capture the cumulative end-state schema as it exists on
main after migrations 000 through 082 run:

- users: drop role, add is_root and status
- user_preference, org_preference: rename preference_id/preference_value
  to name/value
- reset_password_token: add expires_at
- factor_api_key: restructure for service accounts (drop role/token/etc.,
  add name/key/service_account_id)
- drop user_invite (deprecated)
- rename org_domains -> auth_domain
- add user_role, service_account, service_account_role, llm_pricing_rule,
  span_mapper_group, span_mapper, tag, tag_relation
- partial unique indices on users(email,org_id) WHERE status != 'deleted',
  service_account(name/email,org_id) WHERE status != 'deleted',
  cloud_integration(account_id,provider,org_id) WHERE removed_at IS NULL,
  llm_pricing_rule(org_id,source_id) WHERE source_id IS NOT NULL
- update index list to drop user_invite/org_domains entries, add the
  new tables' indices, swap factor_api_key.token for .key

The test asserts the exact SQL output for a fresh-database run.

refs SigNoz/platform-pod#706

therealpandey committed:
feat(sqlmigration): squash 000-082 into s100sqlmigration.V100

Delete the per-step migration files 000_add_data_migrations.go through
082_add_tags.go. NewSQLMigrationProviderFactories now registers just
s100sqlmigration.NewV100Factory, which encodes the cumulative end-state
schema and uses pkg/sqlschema's idempotent CreateTable/AlterTable so it
no-ops on databases that have already applied 000-082.

refs SigNoz/platform-pod#706
